### DISCUSSION TITLE
[QUESTION] Multi-tenancy with different brokers and result backends

### DISCUSSION THREAD

**Discussion by luismojena** [celery/celery#7084] [Q&A]
Hi there, :wave: 
I am working in a multitenant system in which, for compliance (laws and regulations) reasons, each tenant has a separate database, which means that I have to connect to different result backends. 

The app is a flask app that runs in Kubernetes for all tenants, the tenant information is only available when a request hits the flask app, and with that info, we set up a database connection, etc. 

My question is: Is there a way to achieve this in a natural way without having to mess with celery internals? Maybe by providing a callable that returns the connection string?

If this is not supported, what do you recommend doing?
Thanks in advance.

[a comment by an automated account is not shown]

**Comment by teymurov27** [comment]
Hi, if you've solved I am curious how you achieved?
